syntax = "proto2";

package scheduler.protos;

message SchedulerConfig {
  // E.g. 127.0.0.1
  repeated string host_addresses = 1;
  // TF servers need to operate on a port. Per device, a port will be used,
  // Starting from initial_tf_port until initial_tf_port + num_devices_per_worker
  optional uint32 initial_tf_port = 2;

  // In hours; 0 for no limit
  optional float experiment_time_limit = 4;

  // Scheduler updates per second
  optional float ups = 5;
  // In hours; 0 means perform experiment reorganization each update step.
  optional float reorganize_experiments_interval = 6;

  // Path to the resource folder used by experiments
  optional string resource_folder = 7;
  // Path to the resource folder within the docker container
  optional string docker_resource_folder = 8;
}